src/common/load-plugins.js

```diff
@@ -5,31 +5,41 @@ const path = require("path");
 const uniqBy = require("lodash/uniqBy");
 const partition = require("lodash/partition");
 const globby = require("globby");
-const mem = require("mem");
 const internalPlugins = require("../languages.js");
 const thirdParty = require("./third-party.js");
 const resolve = require("./resolve.js");
+const { isNonEmptyArray } = require("./util.js");
+
+const nodeModulesPluginCache = new Map();
+let loadCache = new WeakMap();
 
-const memoizedLoad = mem(load, { cacheKey: JSON.stringify });
-const memoizedSearch = mem(findPluginsInNodeModules);
 const clearCache = () => {
-  mem.clear(memoizedLoad);
-  mem.clear(memoizedSearch);
+  nodeModulesPluginCache.clear();
+  loadCache = new WeakMap();
 };
+const defaultPluginSearchDirs = [];
```

Comment:
This change looks unsafe to me.

Reply:
this is so if we don't pass any plugins, we re-use the same empty array rather than creating a new one each time (which helps keep the caching a lot cleaner). its pretty safe in the current code because we never mutate it, but there is risk someone unaware of this might later mutate it without understanding the side effects.

i could leave a comment to help people understand not to do that